### PR TITLE
client-go: chore: implement `Is` interface for `ErrGroupDiscoveryFailed`

### PR DESCRIPTION
**Pull request by takirala**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

`ErrGroupDiscoveryFailed` implements  `Is` interface so that the callers can use `errors.Is` to compare this error. Using `errors.Is` currently does not work as this https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/errors/wrap.go;l=53 line here is skipped.

This is a common pattern for most other errors, for e.g.: https://github.com/kubernetes/kubernetes/blob/1620473a9a01dd6bbef3398c0acb2e581d0a13c3/staging/src/k8s.io/apimachinery/pkg/api/meta/errors.go#L47-L50

Current implementation of `IsGroupDiscoveryFailedError` https://github.com/kubernetes/kubernetes/blob/1620473a9a01dd6bbef3398c0acb2e581d0a13c3/staging/src/k8s.io/client-go/discovery/discovery_client.go#L420C1-L423 is lacking as the caller has to `Unwrap` (multiple times probably) and then call the `IsGroupDiscoveryFailedError` manually. Adhering to `Is` semantics will make it easier to just invoke `errors.Is(` directly.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/discovery/discovery_client.go [hunk 1]
```diff
@@ -415,6 +415,11 @@ func (e *ErrGroupDiscoveryFailed) Error() string {
 	return fmt.Sprintf("unable to retrieve the complete list of server APIs: %s", strings.Join(groups, ", "))
 }
 
+func (e *ErrGroupDiscoveryFailed) Is(target error) bool {
```
Comment (apelisse) [inline staging/src/k8s.io/client-go/discovery/discovery_client.go:418]: Do you want to add a comment here?

Comment (takirala) [reply staging/src/k8s.io/client-go/discovery/discovery_client.go:419]: updated.

**Comment by takirala** [comment]
I guess I don’t have access to rerun tests but also I think the tests are flakes and should be unaffected by the changes here

**Comment by takirala** [comment]
Can I do anything to take this forward ?

**Pull request merged**